onnxruntime/contrib_ops/cpu/bert/attention_helper.h

```diff
@@ -41,17 +41,15 @@ void ComputeAttentionSoftmaxInplace(T* score, int N, int D, ThreadPool* tp) {
         sum += x[i];
       }
 
-      if (sum == 0) {
-        for (int i = 0; i < D; i++) {
-          y[i] = 1.0f / (float)D;
-        }
-      } else {
-        for (int i = 0; i < D; i++) {
-          y[i] = x[i] / (float)sum;
-        }
+      // `sum` (which we are dividing by below) will never be 0 as long as there is atleast one element in the sequence
+      // (i.e.) D > 0 (sequence_length > 0). If we reach this line of code, we have already ensured that
```

Comment:
Could sum be zero if I had a sequence with two scores of -1 and 1? 

nit: The comment about checking if the input tensor is non-empty is a little unclear to me as there's no check on sequence_len in CheckInputs. Possibly a more direct explanation is that we won't hit this code when called from ComputeAttentionProbs as N and D would both be zero if sequence_len was 0.

Reply:
We subtract the max out which in case would be 1 and hence exp (0) == 1 and exp (-2) is non-negative (as exp’s lower bound is 0) and so the sum cannot be zero.

Reply:
I ll take a look at refining the comment.